lib/battle/DamageCalculator.cpp

```diff
@@ -29,6 +31,15 @@ DamageRange DamageCalculator::getBaseDamageSingle() const
 	minDmg = info.attacker->getMinDamage(info.shooting);
 	maxDmg = info.attacker->getMaxDamage(info.shooting);
 
+    if(minDmg > maxDmg)
+    {
+		const auto & creatureName = info.attacker->creatureId().toEntity(VLC)->getNamePluralTranslated();
+		logGlobal->error("Creature %s min damage %lld exceed max damage %lld.", creatureName, minDmg, maxDmg);
+        logGlobal->error("This problem may lead to unexpected result, please report it to modder.");
+        // to avoid RNG crash. and make bless and curse spell correct
+        std::swap(minDmg, maxDmg);
```

Comment:
```suggestion
	const auto & creatureName = info.attacker->creatureId().toEntity(VLC)->getNamePluralTranslated();
	logGlobal->error("Creature %s: min damage %lld exceeds max damage %lld.", creatureName, minDmg, maxDmg);
        logGlobal->error("This may lead to unexpected results, please report it to the mod's creator.");
        // to avoid an RNG crash and make bless and curse spells work as expected
        std::swap(minDmg, maxDmg);
```